src/test/kotlin/com/igorwojda/integer/factorial/challenge.kt

```diff
@@ -4,7 +4,10 @@ import org.amshove.kluent.shouldBeEqualTo
 import org.junit.jupiter.api.Test
 
 private fun factorial(n: Int): Int {
-    TODO("not implemented")
+    var total = 1
```

Comment:
Re-write this:
1. Using a folding operation
2. Using a recursive algorithm
3. If you're curious, look at tail recursive functions (https://kotlinlang.org/docs/reference/functions.html#tail-recursive-functions). See if you can write factorial as a tailrec function.
Please compare the running time of the solutions.